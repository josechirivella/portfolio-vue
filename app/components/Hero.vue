<template>
  <div
    id="hero"
    class="h-[90svh] flex items-center flex-col justify-center text-lg hero overflow-hidden"
  >
    <h1 class="hero__name">Jose Chirivella</h1>
    <h3 class="hero__title mb-2">Software Engineer</h3>
    <div
      id="icons-container"
      class="mt-4 grid grid-rows-2 lg:grid-rows-1 grid-flow-col gap-4"
    >
      <template v-for="(icon, index) in icons" :key="index">
        <btn-icons :icon="icon.icon" :link="icon.link" />
      </template>
    </div>
  </div>
</template>

<script lang="ts" setup>
import BtnIcons from './BtnIcons.vue';

const icons = [
  {
    link: 'https://twitter.com/chiri14',
    icon: 'fa-brands:twitter',
  },
  {
    link: 'https://linkedin.com/in/josechirivella',
    icon: 'fa-brands:linkedin',
  },
  {
    link: 'https://github.com/josechirivella',
    icon: 'fa-brands:github',
  },
  {
    link: 'https://medium.com/@josechirivella',
    icon: 'fa-brands:medium',
  },
  {
    link: 'mailto:joser.chirivella@gmail.com',
    icon: 'fa6-regular:paper-plane',
  },
  {
    link: 'https://storage.googleapis.com/portfolio-assets/resume/ResumeJose.pdf',
    icon: 'fa6-regular:file-pdf',
  },
];
</script>
