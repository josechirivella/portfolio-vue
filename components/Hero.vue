<template>
  <div
    id="hero"
    class="bg-black text-white min-h-screen flex items-center flex-col justify-center text-lg hero overflow-hidden"
  >
    <h1 class="hero__name">Jose Chirivella</h1>
    <h3 class="hero__title mb-2">Software Engineer</h3>
    <div id="icons-container" class="mt-4 flex align-middle">
      <template v-for="(icon, index) in icons" :key="index">
        <btn-icons :link="icon.link" :icon="icon.icon"></btn-icons>
      </template>
    </div>
  </div>
</template>

<script setup lang="ts">
import { ref } from 'vue';
import BtnIcons from '~/components/BtnIcons';

const icons = ref([
  {
    link: 'https://twitter.com/chiri14',
    icon: 'fa-brands:twitter',
  },
  {
    link: 'https://linkedin.com/in/josechirivella',
    icon: 'fa-brands:linkedin',
  },
  {
    link: 'https://github.com/josechirivella',
    icon: 'fa-brands:github',
  },
  {
    link: 'https://medium.com/@josechirivella',
    icon: 'fa-brands:medium',
  },
  {
    link: 'mailto:joser.chirivella@gmail.com',
    icon: 'fa6-regular:paper-plane',
  },
  {
    link: 'https://storage.googleapis.com/portfolio-assets/resume/ResumeJose.pdf',
    icon: 'fa6-regular:file-pdf',
  },
]);
</script>
