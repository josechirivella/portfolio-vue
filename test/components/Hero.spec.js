import { beforeAll, describe, expect, test } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import Hero from '~/app/components/Hero.vue';

describe('Hero', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallowMount(Hero, {});
  });

  test('should render component', () => {
    expect(wrapper.vm).toBeTruthy();
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
  describe('BtnIcons', () => {
    test('icons length should be greater than 0', () => {
      expect(wrapper.vm.icons).not.toHaveLength(0);
      expect(wrapper.vm.icons).toHaveLength(6);
    });
  });
  describe('Hero content', () => {
    test('name should be present', () => {
      expect(wrapper.find('.hero__name').exists()).toBe(true);
      expect(wrapper.find('.hero__name').text()).toEqual('Jose Chirivella');
    });
    test('title should be present', () => {
      expect(wrapper.find('.hero__title').exists()).toBe(true);
      expect(wrapper.find('.hero__title').text()).toEqual('Software Engineer');
    });
  });
});
