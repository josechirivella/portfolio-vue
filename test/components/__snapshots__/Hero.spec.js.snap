// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero should render component 1`] = `
<div
  class="bg-black text-white min-h-screen flex items-center flex-col justify-center text-lg hero overflow-hidden"
  id="hero"
>
  <h1
    class="hero__name"
  >
    Jose Chirivella
  </h1>
  <h3
    class="hero__title mb-2"
  >
    Software Engineer
  </h3>
  <div
    class="mt-4 flex align-middle"
    id="icons-container"
  >
    
    <btn-icons-stub
      icon="fa-brands:twitter"
      link="https://twitter.com/chiri14"
    />
    <btn-icons-stub
      icon="fa-brands:linkedin"
      link="https://linkedin.com/in/josechirivella"
    />
    <btn-icons-stub
      icon="fa-brands:github"
      link="https://github.com/josechirivella"
    />
    <btn-icons-stub
      icon="fa-brands:medium"
      link="https://medium.com/@josechirivella"
    />
    <btn-icons-stub
      icon="fa6-regular:paper-plane"
      link="mailto:joser.chirivella@gmail.com"
    />
    <btn-icons-stub
      icon="fa6-regular:file-pdf"
      link="https://storage.googleapis.com/portfolio-assets/resume/ResumeJose.pdf"
    />
    
  </div>
</div>
`;
